### PR TITLE
Windows: Add CI for per-commit/release image build

### PR DESCRIPTION
**Pull request by sunjayBhatia**
Commit Message:
Windows: Add CI for per-commit/release image build

- Make ./ci/docker_ci.sh cross platform
- Add CI job to build image from result of build and test CI job
- Add Docker image built in same structure as Linux version, mirrored entrypoint/command
  - google proxy example config used like in Linux build
- Docker image base can be overridden with different Windows container image if needed

Additional Description: N/A
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
cc @envoyproxy/windows-dev

**Comment by sunjayBhatia** [comment]
Hey @lizan did you have any thoughts about this change? Do we need to create the appropriate repos in the envoyproxy Docker hub org or does the CI user have permissions to create new repos?

File: ci/docker_ci.sh [hunk 1]
```diff
@@ -40,12 +48,15 @@ build_args() {
 }
 
 use_builder() {
+# BuildKit is not available for Windows images, skip this
+if ! is_windows; then
```
Comment (lizan) [inline ci/docker_ci.sh:52]: nit: indent

File: ci/docker_ci.sh [hunk 2]
```diff
@@ -20,11 +24,15 @@ build_platforms() {
   TYPE=$1
   FILE_SUFFIX="${TYPE/-debug/}"
 
+if is_windows; then
+  echo "windows/amd64"
+else
```
Comment (lizan) [inline ci/docker_ci.sh:29]: `elif`?

**Review by lizan — COMMENTED** [review]
some shell nits, thanks

**Review by lizan — APPROVED** [review]

**Pull request merged**